+nosnoc/+discrete_time_problem/Heaviside.m

```diff
@@ -35,6 +35,10 @@ function create_variables(obj)
             obj.w.v_global = {{'v_global',dims.n_v_global}, model.lbv_global, model.ubv_global, model.v0_global};
             if opts.use_speed_of_time_variables && ~opts.local_speed_of_time_variable
                 obj.w.sot = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+                if opts.time_freezing
+                    obj.p.rho_sot = {{'rho_sot',1}, opts.rho_sot};
+                    obj.f = obj.f + obj.p.rho_sot()*(obj.w.sot()-1)^2;
```

Comment:
should the sot variable be different for every control stage? otherwise, the problem might be infeasible.

Reply:
in other words, we should not have local sot in tf... god this is complicated :D 
I suggest to throw an error here and inform the user to use `local_speed_of_time_variable = true` (which is default anyway)  if `time_freezing = true`.

Reply:
You mean we should not have global sot in tf? I think we _need_ local sot in order to maintain feasibility in each control stage when `stagewise_clock_constraint`/`equidistant_control_grid` is true.

Reply:
indeed, i confused myself a bit. global is needed e.g. in simulation, as there is only one stage (but it is equivalent to local).
local is needed for `stagewise_clock_constraint =true`. so we on the other hand, global sot with multiple control stages might lead to infeasible problems. 

in summary i was questioning the need of global sot, as local does the job (and my be even necessary). think of two stage, one has a jump and the other doesnt, and stagewise clock constraint. it can be feasible only with local.